core/src/com/unciv/logic/map/mapgenerator/NaturalWonderGenerator.kt

```diff
@@ -22,410 +24,199 @@ class NaturalWonderGenerator(val ruleset: Ruleset, val randomness: MapGeneration
-        }
 
-        trySpawnOnSuitableLocation(suitableLocations, wonder)
-    }
+    Mt. Fuji: Must be in grass or plains; avoids oceans and the biggest landmass; cannot be adjacent to tundra,
+    desert, marsh, or mountains;can be adjacent to a maximum of 2 hills; becomes mountain
 
-    /*
     Great Barrier Reef: Specifics currently unknown;
     Assumption: at least 1 neighbour coast; no tundra; at least 1 neighbour coast; becomes coast
-    */
-    private fun spawnGreatBarrierReef(tileMap: TileMap) {
-        val wonder = ruleset.terrains[Constants.greatBarrierReef]!!
-        val suitableLocations = tileMap.values.filter {
-            it.resource == null
-                    && wonder.occursOn.contains(it.getLastTerrain().name)
-                    && abs(it.latitude) > tileMap.maxLatitude * 0.1
-                    && abs(it.latitude) < tileMap.maxLatitude * 0.7
-                    && it.neighbors.any { it.baseTerrain == Constants.coast }
-                    && it.neighbors.all { neighbor -> neighbor.isWater }
-                    && it.neighbors.any { neighbor ->
-                neighbor.resource == null && neighbor.improvement == null
-                        && wonder.occursOn.contains(neighbor.getLastTerrain().name)
-                        && neighbor.neighbors.all { it.isWater }
-            }
-        }
 
-        val location = trySpawnOnSuitableLocation(suitableLocations, wonder)
-        if (location != null) {
-            val possibleLocations = location.neighbors
-                    .filter {
-                        it.resource == null
-                                && wonder.occursOn.contains(it.getLastTerrain().name)
-                                && it.neighbors.all { it.isWater }
-                    }.toList()
-            trySpawnOnSuitableLocation(possibleLocations, wonder)
-        }
-    }
-
-    /*
     Krakatoa: Must spawn in the ocean next to at least 1 shallow water tile; cannot be adjacent
     to ice; changes tiles around it to shallow water; mountain
-    */
-    private fun spawnKrakatoa(tileMap: TileMap) {
-        val wonder = ruleset.terrains[Constants.krakatoa]!!
-        val suitableLocations = tileMap.values.filter {
-            it.resource == null
-                    && wonder.occursOn.contains(it.getLastTerrain().name)
-                    && it.neighbors.any { neighbor -> neighbor.getBaseTerrain().name == Constants.coast }
-                    && it.neighbors.none { neighbor -> neighbor.getLastTerrain().name == Constants.ice }
-        }
-
-        val location = trySpawnOnSuitableLocation(suitableLocations, wonder)
-        if (location != null) {
-            for (tile in location.neighbors) {
-                if (tile.baseTerrain == Constants.coast) continue
-                tile.baseTerrain = Constants.coast
-                clearTile(tile)
-            }
-        }
-    }
 
-    /*
     Rock of Gibraltar: Specifics currently unknown
```

Comment:
Based on the source code of the original:
- Spawns on land tiles
- Requires at least 3 adjacent coast tiles
- Requires exactly 1 adjacent land tile

Reply:
OK, but I just migrated the existing base -> _now_ you can fix that by json only

Reply:
Agree, just something I came across so feel free to ignore this for now